edgelet/edgelet-utils/src/lib.rs

```diff
@@ -59,7 +59,19 @@ const DNS_MAX_SIZE: usize = 63;
 ///  - must start with an alphabet
 ///  - must end with an alphanumeric character
 pub fn sanitize_dns_label(name: &str) -> String {
-    name.trim_start_matches(|c: char| !c.is_ascii_alphabetic())
+    sanitize_dns_label_with(name, char::is_ascii_alphabetic)
+}
+
+/// The name returned from here must conform to following rules (as per RFC 1035/1123):
+///  - length must be <= 63 characters
+///  - must be all lower case alphanumeric characters or '-'
+///  - must start and end with an alphanumeric character
```

Comment:
do we need to catch any exception if the input does not adhere this standard?

Reply:
This function should create a conforming string or an empty string.  Some times empty strings are OK, like in label values. but when empty string are not they're caught in this function: edgelet/edgelet-kube/src/convert/mod.rs:27